clvm/run_program.py

```diff
@@ -57,16 +57,39 @@ def traverse_path(sexp: SExp, env: SExp) -> Tuple[int, SExp]:
         cost = PATH_LOOKUP_COST_PER_LEG
         if sexp.nullp():
             return cost, sexp.null()
-        node_index = int_from_bytes(sexp.atom)
-        while node_index > 1:
+
+        b = sexp.atom
+
+        end_byte_cursor = 0
+        while end_byte_cursor < len(b) and b[end_byte_cursor] == 0:
+            end_byte_cursor += 1
+
+        cost += end_byte_cursor * PATH_LOOKUP_COST_PER_ZERO_BYTE
+        if end_byte_cursor == len(b):
+            return cost, sexp.null()
+
+        # create a bitmask for the most significant *set* bit
+        # in the last non-zero byte
```

Comment:
Wow, how does this work? Is this from https://graphics.stanford.edu/~seander/bithacks.html ?

Reply:
yes, inspired by: https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2

Reply:
there really should be a unit test for this, since it's not obviously correct by inspection